.github/workflows/delegator_generic_slsa3.yml

```diff
@@ -0,0 +1,257 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA builder delegator
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  SLSA_OUTPUTS_DIR: __SLSA_OUTPUTS_DIR__
+  SLSA_ARTIFACTS_FILE: artifacts-layout.json
+  SLSA_PREDICATE_FILE: predicate.json
+
+on:
+  workflow_call:
+    # Secrets provided as arguments are passed as inputs to the callback Action.
+    secrets:
+      secret1:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+      secret2:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+
+    # Inputs provided as arguments are passed as inputs to the callback Action,
+    # formatted as a map.
+    inputs:
+      slsa-token:
+        description: "The SLSA token identifying the request"
+        required: true
+        type: string
+
+    # Outputs are provided to the caller TRW.
+    outputs:
+      # This contains all the ouputs of the callback Action, formatted as a map.
+      build-artifacts-outputs:
+        description: "The outputs from the build-artitacts Action, unchanged."
+        value: ${{ jobs.build-artifacts-ubuntu.outputs.outputs }}
+      
+      # This is an output from the framework.
+      attestations-download-name:
+        description: "Name of the artifact to download all the attestations."
+        value: ${{ jobs.generate-provenance.outputs.attestations-download-name }}
+
+jobs:
+  # rng generates a random number to avoid name collision in artifacts
+  # when multiple workflows run concurrently.
+  rng:
+    outputs:
+      value: ${{ steps.rng.outputs.random }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate random 16-byte value (32-char hex encoded)
+        id: rng
+        uses: slsa-framework/slsa-github-generator/.github/actions/rng@main
+
+  # verify-token verifies the slsa token.
+  verify-token:
+    outputs:
+      slsa-verified-token: ${{ steps.verify.outputs.slsa-verified-token }}
+      tool-repository: ${{ steps.verify.outputs.tool-repository }}
+      tool-ref: ${{ steps.verify.outputs.tool-ref }}
+      predicate-sha256: ${{ steps.upload.outputs.sha256 }}
+    runs-on: ubuntu-latest
+    needs: [rng]
+    steps:
+      - name: Verify token with test action
+        id: verify
+        uses: slsa-framework/slsa-github-generator/.github/actions/verify-token@main
+        with:
+          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+          slsa-unverified-token: ${{ inputs.slsa-token }}
+          output-predicate: ${{ env.SLSA_PREDICATE_FILE }}
+
+      - name: Debug output predicate
```

Comment:
since we upload the predicate i think we can remove the debug step

Reply:
done. I've kept a few other debug to make sure we can quickly fix problems as we start building the builders.